pkg/operator/v1alpha1helpers/helpers.go

```diff
@@ -207,3 +210,38 @@ func EnsureOperatorConfigExists(client dynamic.Interface, operatorConfigBytes []
 		}
 	}
 }
+
+type InstallConfigGetter func() (installertypes.InstallConfig, error)
+
+// InstallConfigFromFile gets the install config from the cluster configmap file data
+func InstallConfigFromFile(configMapData []byte) InstallConfigGetter {
+	return func() (installertypes.InstallConfig, error) {
+		obji, err := runtime.Decode(scheme.Codecs.UniversalDecoder(corev1.SchemeGroupVersion), configMapData)
+		if err != nil {
+			return installertypes.InstallConfig{}, err
+		}
+		cm, ok := obji.(*corev1.ConfigMap)
+		if !ok {
+			return installertypes.InstallConfig{}, fmt.Errorf("expected *corev1.ConfigMap found %T", obji)
+		}
+
+		return InstallConfigFromClusterConfig(cm)
+	}
+}
+
+// InstallConfigFromClusterConfig gets the install config from the cluster configmap
+func InstallConfigFromClusterConfig(configMap *corev1.ConfigMap) (installertypes.InstallConfig, error) {
+	var (
+		icKey = "install-config"
+		ic    installertypes.InstallConfig
```

Comment:
it's more Go-like to declare variable directly where they are needed instead of at the top of the func.